WordPress/gutenberg
# User Patterns: only add core pattern categories to user taxonomy when they are used

## What?
Instead of adding all the core pattern categories to the new wp_pattern taxonomy, only add them as they are assigned.

## Why?
In https://github.com/WordPress/gutenberg/pull/51144 adding all the pattern categories to the user taxonomy and relying on them being there causes issues with users not being able to delete them, and locking down the taxonomy removes the flexibility of users being able to add their own terms

## How?
Merges the user taxonomy with core patterns and adds a new term if a core category is selected and it has no taxonomy id

## Testing Instructions

- Go to `/wp-admin/edit-tags.php?taxonomy=wp_pattern` and delete any existing categories, and add one or more new categories that don't exist in core pattern categories
- Go to post editor and add a block and select option in overflow menu to `Create pattern`
- You should see the core pattern categories and the custom ones you added
- If you select a core category a new taxonomy term should be added

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/3629020/75fabce3-e49b-45b1-9031-313f4c20794a
